underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -105,56 +96,59 @@ public int read(byte[] b) throws IOException {
 
   @Override
   public int read(byte[] b, int offset, int length) throws IOException {
-    if (length == 0) {
-      return 0;
+    S3Object object = getObject(mPos, mPos + length);
+    if (object == null) {
+      // range request cannot meet
+      object = getObject(mPos);
```

Comment:
could you elaborate what's the case this request could not meet. I fee like this might be covered by precondition check

Reply:
need to do more testing to understand, that's from their getObject() API description. If not meet request, will return null. They may return null for different reasons (e.g. wrong bucket name, wrong key)